ios/OktaSdkBridge.swift

```diff
@@ -13,14 +13,74 @@
 import Foundation
 import OktaOidc
 
+// MARK: - OktaOidcProtocol
+
+protocol OktaOidcProtocol: class {
+    var configuration: OktaOidcConfig { get }
+    
+    func signInWithBrowser(from presenter: UIViewController,
+                           additionalParameters: [String : String],
+                           callback: @escaping ((OktaOidcStateManager?, Error?) -> Void))
+    
+    func signOutOfOkta(_ authStateManager: OktaOidcStateManager,
+                       from presenter: UIViewController,
+                       callback: @escaping ((Error?) -> Void))
+    
+    func authenticate(withSessionToken sessionToken: String,
+                      callback: @escaping ((OktaOidcStateManager?, Error?) -> Void))
+}
+
+extension OktaOidc: OktaOidcProtocol {
+    
+}
+
+// MARK: - StateManagerProtocol
+
+protocol StateManagerProtocol: class {
+    var accessToken: String? { get }
+    var idToken: String? { get }
+    var refreshToken: String? { get }
+
+    func getUser(_ callback: @escaping ([String:Any]?, Error?) -> Void)
+    func renew(callback: @escaping ((OktaOidcStateManager?, Error?) -> Void))
+    func revoke(_ token: String?, callback: @escaping (Bool, Error?) -> Void)
+    func introspect(token: String?, callback: @escaping ([String : Any]?, Error?) -> Void)
+    func clear()
+}
+
+extension OktaOidcStateManager: StateManagerProtocol {
+    
+}
+
+// MARK: - OktaSdkBridge
+
 @objc(OktaSdkBridge)
 class OktaSdkBridge: RCTEventEmitter {
+    var config: OktaOidcConfig? {
+        oktaOidc?.configuration
+    }
     
-    private var oktaOidc: OktaOidc?
-    private var config: OktaOidcConfig?
+    private(set) var oktaOidc: OktaOidcProtocol?
     
     override var methodQueue: DispatchQueue { .main }
     
+    init(oidc: OktaOidcProtocol? = nil) {
+        self.oktaOidc = oidc
+    }
+    
+    func presentedViewController() -> UIViewController? {
+        RCTPresentedViewController()
+    }
+    
+    func readStateManager() -> StateManagerProtocol? {
```

Comment:
Perhaps this could be renamed to be a var with a computed getter? e.g.

```
var stateManager: StateManagerProtocol? {
  ...
}
```

Reply:
Good input. Done.